lib/core-integration/src/Test/Integration/Scenario/API/Transactions.hs

```diff
@@ -1025,26 +1043,338 @@ spec = do
+                  , [ expectListSizeEqual 2
+                    , expectListItemFieldEqual 0 amount a2
+                    , expectListItemFieldEqual 1 amount a1
+                    ]
+                  )
+                -- 11
+                , ( "?start="
+                    <> (toParam $ addUTCTime (-1) t1)
+                    <> "&end="
+                    <> (toParam t2)
+                  , [ expectListSizeEqual 2
+                    , expectListItemFieldEqual 0 amount a2
+                    , expectListItemFieldEqual 1 amount a1
+                    ]
+                  )
+                -- 12
+                , ( "?start="
+                    <> (toParam $ addUTCTime (-1) t1)
+                    <> "&end="
+                    <> (toParam $ addUTCTime 1 t2)
+                  , [ expectListSizeEqual 2
+                    , expectListItemFieldEqual 0 amount a2
+                    , expectListItemFieldEqual 1 amount a1
+                    ]
+                  )
+                -- 13
+                , ( mempty
+                  , [ expectListSizeEqual 2
+                    , expectListItemFieldEqual 0 amount a2
+                    , expectListItemFieldEqual 1 amount a1
+                    ]
+                  )
+                -- 14
+                , ( "?end="
+                    <> (toParam t2)
+                  , [ expectListSizeEqual 2
+                    , expectListItemFieldEqual 0 amount a2
+                    , expectListItemFieldEqual 1 amount a1
+                    ]
+                  )
+                -- 15
+                , ( "?end="
+                    <> (toParam $ addUTCTime 1 t2)
+                  , [ expectListSizeEqual 2
+                    , expectListItemFieldEqual 0 amount a2
+                    , expectListItemFieldEqual 1 amount a1
+                    ]
+                  )
+                -- 16
+                , ( "?end="
+                    <> (toParam $ addUTCTime (-1) t2)
+                  , [ expectListSizeEqual 1
+                    , expectListItemFieldEqual 0 amount a1
+                    ]
+                  )
+                ]
+
+        forM_ matrix $ \(query, expectations) -> do
+          rf <- request @([ApiTransaction t]) ctx (listTxEp w query)
+              Default Empty
+          verify rf expectations
+
+    describe "TRANS_LIST_02,03 - Faulty start, end, order values" $ do
+        let orderErr = "Please specify one of the following values:\
+            \ ascending, descending."
+        let startEndErr = "Expecting ISO 8601 date-and-time format\
+            \ (basic or extended), e.g. 2012-09-25T10:15:00Z."
+        let queries =
+                  [ ( "?start=2009"
+                    , [ expectResponseCode @IO HTTP.status400
+                      , expectErrorMessage startEndErr
+                      ]
+                    )
+                  ,
+                    ( "?start=2012-09-25T10:15:00Z&end=2016-11-21"
```

Comment:
I suspect that the `:` character should be URL-encoded here.

Perhaps a good use case for the `toQueryString` function? (This handles URL encoding automatically.)

https://github.com/input-output-hk/cardano-wallet/pull/613/files#diff-213a7e10be74fd4570beffc3464fd728R825